solutions/communication_kafka/README.md

```diff
@@ -0,0 +1,423 @@
+>The Marker interface, part of the org.slf4j package, renders the FATAL level largely redundant.
+> -- <cite>https://www.slf4j.org/faq.html#fatal</cite>
+
+It is also dropped due to the fact that we should not determine when to terminate an application in a logging framework (https://www.baeldung.com/slf4j-with-log4j2-logback).
+
+
+#### Tracing
+In microservice architecture, tracing is implemented to monitor applications as well as to help identify where errors or failures occur, which may cause poor performance. Just like in our application which contains several services, it is necessary to trace the invocation from one service to another, either directly or through a kafka message broker.
+
+In this sample application, tracing implementations are supported by Spring Cloud Sleuth. Sleuth seamlessly interfaces with logging frameworks such as SLF4J and logback to add unique IDs that aid in the tracking and diagnosis of issues leveraging logs. Before it is implemented, its dependency must be defined in `build.gradle` file as following:
+```
+dependencyManagement {
+	imports {
+		mavenBom "org.springframework.cloud:spring-cloud-dependencies:2021.0.2"
+	}
+}
+
+dependencies {
+    implementation 'org.springframework.cloud:spring-cloud-starter-sleuth'
+}
+```
+Once added within the application, Spring Cloud Sleuth automatically formats the logs that contain traceId and spanId. Below you can see the logs when the application just started and when a customer just booked a ship.
+
+```
+2022-06-13 22:01:16.243  INFO [shipkafka,bd2025db4480982a,bd2025db4480982a] 14272 --- [nio-8080-exec-5] o.a.kafka.common.utils.AppInfoParser     : Kafka startTimeMs: 1655150476243
+```
+```
+2022-06-13 22:01:16.430  INFO [shipkafka,bd2025db4480982a,cfcad1e77644ff13] 14272 --- [ bookings-0-C-1] c.d.s.s.api.ShipRestController           : Received: Booking(createdOn=Mon Jun 13 22:01:16 CEST 2022, id=12, lastUpdatedOn=Mon Jun 13 22:01:16 CEST 2022, containerCount=2, shipId=4, bookingStatus=REQUESTED, version=0)
+```
+The part of normal log with additional core information from Spring Sleuth follows the format of:
+**[application name, traceId, spanId]**
+
+- **Application name** - name that is set in the properties file/settings.gradle file. It can be used to aggregate logs from multiple instances of the same application.
+- **traceId** - unique identifier for a specific job, request, or action. It is the same across all microservices.
+- **spanId** - unique identifier that tracks a unit of work. It is assigned to each operation and therefore can be vary depends on what request is performed.
+
+
+Once the application is started, the traceID and spanID **will be the same** by default.   
+
+Further reading: https://www.baeldung.com/spring-cloud-sleuth-single-application#:~:text=TraceId%20%E2%80%93%20This%20is%20an%20id,that%20consists%20of%20multiple%20steps.
+
+#### Retry (in progress...)
```

Comment:
This would fit good into the sample application documentation.

In the solution the documentation should be limited in the last step on the how in a generic way.
1. How to do monitoring: -> Here it should only be a code snippet with a short explanation
2. How to do logging: -> Here the same
3. How to use the retry mechanism: -> Code snippet if necessary (or just the configuration parts explained)